kore/src/main/scala/org/kframework/parser/TreeNodesToKORE.scala

```diff
@@ -29,21 +29,23 @@ class TreeNodesToKORE(parseSort: java.util.function.Function[String, Sort], stri
     if (strict) KApply(lbl, KList(KToken("_", Sorts.KVariable)), Att.add(classOf[Production], Production(lbl, Seq(), sort, Seq(NonTerminal(sort, None))))) else KToken("_", Sorts.KVariable)
   }
 
-  def termConsToKApply(t: TermCons, items: PStack[Term], p: Production): KApply = {
-    if (p.att.contains("recordPrd", classOf[Production])) {
-      val realProd = p.att.get("recordPrd", classOf[Production])
-      val map = new util.ArrayList(items).asScala.reverse.zipWithIndex.map { case (item, idx) => (p.nonterminal(idx).name.get, apply(item))} toMap
+  def termConsToKApply(t: TermCons): K = {
+    if (t.production.att.contains("recordPrd", classOf[Production])) {
+      val realProd = t.production.att.get("recordPrd", classOf[Production])
+      val map = t.items.asScala.reverse.zipWithIndex.map { case (item, idx) => (t.production.nonterminal(idx).name.get, apply(item))} toMap
```

Comment:
This needs to be converted to an arraylist still because otherwise reverse will take quadratic time.